File: ReportNotation/src/main/java/osp/leobert/android/reportprinter/notation/Bug.java
package osp.leobert.android.reportprinter.notation;

/**
 * <p><b>Package:</b> osp.leobert.android.reportprinter.notation </p>
 * <p><b>Project:</b> ReportPrinter </p>
 * <p><b>Classname:</b> Bug </p>
 * <p><b>Description:</b> TODO </p>
 * Created by leobert on 2018/7/7.
 */
public @interface Bug {
    State state() default State.TODO;

    String desc();
}

File: m_review_reporter/src/main/java/osp/leobert/android/reporter/review/Bug.kt
package osp.leobert.android.reporter.review

/**
 * Created by leobert on 2021/9/7.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.TYPE_PARAMETER,
    AnnotationTarget.PROPERTY,
    AnnotationTarget.FIELD,
    AnnotationTarget.LOCAL_VARIABLE,
    AnnotationTarget.VALUE_PARAMETER,
    AnnotationTarget.CONSTRUCTOR,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.TYPE,
//    AnnotationTarget.EXPRESSION,
//    AnnotationTarget.FILE,
//    AnnotationTarget.TYPEALIAS
)
annotation class Bug(
    val desc: String,
    val module: String,
    val track: String
)
